### PR TITLE
Validate application snapshots images

### PR DESCRIPTION
**Pull request by caugello**
Verifies for each containerImage:
- signature
- attestation
- EC-policy

Closes HACBS-531 HACBS-532 HACBS-534

### CODE REVIEW

File: cmd/eval.go [hunk 1]
```diff
@@ -83,9 +85,78 @@ func evalCmd() *cobra.Command {
...
+
+				for _, c := range snapshotSpec.Components {
+					wg.Add(1)
+
+					go func(comp appstudioshared.ApplicationSnapshotComponent) {
+						defer wg.Done()
+
+						out, err := validateImage(comp.ContainerImage)
+						cherr <- err
+						errs = append(errs, <-cherr)
```
Comment (zregvart) [inline cmd/eval.go:138]: This looks odd, shouldn't we fetch from the channel outside of the gorutine?

File: cmd/eval.go [hunk 2]
```diff
@@ -96,39 +167,18 @@ func evalCmd() *cobra.Command {
...
-				out.SetAttestationSignatureCheck(false, err.Error())
-				return err
-			}
-			out.SetAttestationSignatureCheck(true, "success")
-
-			p, err := policy.NewPolicyEvaluator(arguments.policyConfiguration)
-			if err != nil {
-				return err
+				if arguments.strict && !success {
+					os.Exit(1)
```
Comment (zregvart) [inline cmd/eval.go:173]: Let's return error instead, error handling in [HACBS-633](https://issues.redhat.com/browse/HACBS-633) should take care of exiting with the proper status

File: cmd/eval.go [hunk 3]
```diff
@@ -83,9 +85,78 @@ func evalCmd() *cobra.Command {
...
+				results, err := p.Evaluate(cmd.Context(), i.Attestations())
+				if err != nil {
+					return nil, err
+				}
+				out.SetPolicyCheck(results)
+
+				return out, nil
+			}
+
+			if snapshotSpec != nil {
```
Comment (zregvart) [inline cmd/eval.go:122]: Something to refactor in a followup PR, we could create an artificial ApplicationSnapshot when passing the image via `--image` and here assume we were given ApplicationSnapshot

File: internal/applicationsnapshot/report_test.go [hunk 4]
```diff
@@ -61,7 +62,21 @@ func Test_Report(t *testing.T) {
...
+		components = append(components, c)
+	}
 
-	report, _ := Report(snapshot)
+	report, _, success := Report(components)
 	assert.JSONEq(t, expected, report)
+	assert.True(t, success)
+
+	components = append(components, Component{Success: false})
+	report, _, success = Report(components)
```
Comment (zregvart) [inline internal/applicationsnapshot/report_test.go:80]: I guess we forgot to assert the `report` here?

**Review by zregvart — COMMENTED** [review]
I think we're blocking in the goroutine by fetching the value from the channel

File: internal/applicationsnapshot/report_test.go (hunk 4 above)
Comment (caugello) [reply internal/applicationsnapshot/report_test.go:111]: Yup, added

File: cmd/eval.go (hunk 1 above)
Comment (caugello) [reply cmd/eval.go]: Yes, moved it.

File: cmd/eval.go [hunk 5]
```diff
@@ -83,9 +85,78 @@ func evalCmd() *cobra.Command {
...
+						cherr <- err
+						errs = append(errs, <-cherr)
+
+						image := applicationsnapshot.Component{
+							Violations: out.PolicyCheck,
+							Success:    out.ExitCode == 0,
+						}
+						image.Name, image.ContainerImage = comp.Name, comp.ContainerImage
+						component <- image
+						components = append(components, <-component)
```
[the remark this answers is not in the document]
Comment (caugello) [reply cmd/eval.go]: Moved it.

File: cmd/eval.go (hunk 2 above)
Comment (caugello) [reply cmd/eval.go]: Yes, added with TODO as reminder.

**Review by zregvart — APPROVED** [review]
:sunglasses:

**Pull request merged**